app/models/operatingsystems/junos.rb

```diff
@@ -25,7 +25,7 @@ def pxedir
   end
 
   def boot_filename(host = nil)
-    "ztp.cfg/" + host.mac.delete(':').upcase
+    "ztp.cfg/" + host.mac.delete(':').upcase + ".cfg"
```

Comment:
Please explain. JunOS is something different than Huawei thing, is it? Is this related? And why are you changing this? Note this is a breaking change.

Reply:
Hey @lzap , in details it's different but they share similarities. In my smart-proxy patch I've also added the .cfg so it works. JunOS does not seem to picky on the name, however, Huawei VRP seems to not consider the configuration file if this extension is missing (as VRP supports other means as well). It didn't look so dangerous to add the extension; however, if you prefer, I can add the extension only if it's a huawei switch :-)

Reply:
That's fine. Thanks.

Reply:
Hey @lzap, just to be sure, you want me to add the code which only adds a .cfg for huawei switches (and thus remove the general addition of this extension for all ZTP switches), right? :-)

Reply:
Removed the ".cfg"-suffix from JunOS :-)